### PR TITLE
Improved pages design and added friend search functionality

### PR DESCRIPTION
**Pull request by Taanviir**

### CODE REVIEW

**Review by PaulaBalbino — APPROVED** [review]

File: backend/static/scripts/pong.js [hunk 1]
```diff
@@ -1247,7 +1247,8 @@ function setupDropdownListeners() {
 
 function visual() {
     if (game.isGameRunning() == true) {
-        var myModal = new bootstrap.Modal('#visualmodal');
+        game.pause();
```
Comment (PaulaBalbino) [inline backend/static/scripts/pong.js:1250]: When imp mode is pressed during game I understand that we do not need to pause the game, only display a message indicating that this is not allowed. If you check the main branch you will notice the ball is not stopping.
We don't need this line. 1250

**Review by PaulaBalbino — CHANGES_REQUESTED** [review]
Please check the comments regarding the imp mode

File: backend/static/scripts/pong.js (hunk 1 above)
Comment (Taanviir) [reply backend/static/scripts/pong.js:1250]: isnt that problematic? the game keeps proceeding and then it can even end

Comment (PaulaBalbino) [reply backend/static/scripts/pong.js:1250]: If that is the case, what we need to do is include the resume similarly to what we did for the configuration (applyConfiguration function). If you prefer to follow this approach I can pause/resume accordinly.

File: backend/templates/pong/pong.html [hunk 2]
```diff
@@ -92,12 +92,14 @@
 </div>
 
 <!-- Visual Imp Mode Modal -->
-<div id="visualmodal" class="modal fade" tabindex="-1">
+<div id="visualmodal" class="modal fade" data-bs-backdrop="static" tabindex="-1">
```
Comment (Taanviir) [inline backend/templates/pong/pong.html:95]: please add 
```html
data-bs-keyboard="false"
```

to prevent keyboard input from disabling modal

**Pull request merged**